### DISCUSSION TITLE
Not getting Light (pup diveces) to work with move hub

### DISCUSSION THREAD

**Discussion by JobadSayas** [pybricks/support#951] [General]
Hi, I am trying to get the light working in a move hub. I am actually using the simple example in the documentation:

```python
from pybricks.pupdevices import Light
from pybricks.parameters import Port
from pybricks.tools import wait

# Initialize the light.
light = Light(Port.D)

# Blink the light forever.
while True:
    # Turn the light on at 100% brightness.
    light.on(100)
    wait(500)

    # Turn the light off.
    light.off()
    wait(500)
```

Notice that the example has Port as A, I set it to D since that's where I have my Light connected. But when I run it, the motor B is the one that moves, then waits 500, then moves, then wait... why? doesn't make any sence.

Am I doing something wrong? is the light compatible with move hub at all?

Please help, I don't get much information about this topic on the internet nor youtube.
Regards
Jobad Sayas

**Comment by dlech** [comment]
This is a bug and was already reported in https://github.com/pybricks/support/issues/913 and has been fixed. The fix hasn't been officially released yet. If you don't want to wait for the next release, you can grab the firmware from that issue and use it.
